src/Cli/dotnet/commands/dotnet-new/PostActions/DotnetAddJsonPropertyPostActionProcessor.cs

```diff
@@ -0,0 +1,235 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.Encodings.Web;
+using System.Text.Json;
+using System.Text.Json.Nodes;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.TemplateEngine.Abstractions;
+using Microsoft.TemplateEngine.Abstractions.PhysicalFileSystem;
+using Microsoft.TemplateEngine.Cli.PostActionProcessors;
+
+namespace Microsoft.DotNet.Tools.New.PostActionProcessors
+{
+    internal class DotnetAddJsonPropertyPostActionProcessor : PostActionProcessorBase
+    {
+        public override Guid Id => ActionProcessorId;
+
+        internal static Guid ActionProcessorId { get; } = new Guid("695A3659-EB40-4FF5-A6A6-C9C4E629FCB0");
+
+        private static readonly JsonSerializerOptions s_serializerOptions = new JsonSerializerOptions
+        {
+            Encoder = JavaScriptEncoder.UnsafeRelaxedJsonEscaping,
+            WriteIndented = true
+        };
+
+        private static readonly JsonDocumentOptions s_deserializerOptions = new JsonDocumentOptions
+        {
+            AllowTrailingCommas = true,
+            CommentHandling = JsonCommentHandling.Skip
+        };
+
+        private const string JsonFileNameArgument = "jsonFileName";
+        private const string ParentPropertyPathArgument = "parentPropertyPath";
+        private const string NewJsonPropertyNameArgument = "newJsonPropertyName";
+        private const string NewJsonPropertyValueArgument = "newJsonPropertyValue";
+
+        protected override bool ProcessInternal(
+            IEngineEnvironmentSettings environment,
+            IPostAction action,
+            ICreationEffects creationEffects,
+            ICreationResult templateCreationResult,
+            string outputBasePath)
+        {
+            if (!action.Args.TryGetValue(JsonFileNameArgument, out string? jsonFileName))
+            {
+                Reporter.Error.WriteLine(string.Format(LocalizableStrings.PostAction_ModifyJson_Error_ArgumentNotConfigured, JsonFileNameArgument));
+                return false;
+            }
+
+            IReadOnlyList<string> jsonFiles = FindFilesAtOrAbovePath(environment.Host.FileSystem, outputBasePath, matchPattern: jsonFileName, maxAllowedAboveDirectories: 1);
```

Comment:
imo, this is a misuse of this method as json is expected exactly on path.
the refactoring to two different methods with extraction of common functionality may be of benefit.